pandas/io/common.py

```diff
@@ -125,6 +125,12 @@ def _validate_header_arg(header) -> None:
             "the row(s) making up the column names"
         )
 
+    # In version 0.24, this would have worked.
+    if header == -1:
```

Comment:
this should only validate if header is an int (or it could be an array-like of ints); these should raise if < 0 right?

Reply:
@jreback : I agree that this check needs to be expanded as you described

Reply:
The test that was added will of course need to be updated as well

Reply:
Sure - thank you all for having taken the time to review my work!